writeengine/wrapper/writeengine.cpp

```diff
@@ -953,6 +1130,58 @@ static void log_this(const char *message,
 }
 #endif
 
+/** @brief Determine whether we may update a column's ranges (by type) and return NULL if we can't */
+static BRM::CPInfo*
+getCPInfoToUpdateForUpdatableType(const ColStruct& colStruct, BRM::CPInfo* currentCPInfo)
+{
+    switch(colStruct.colType)
+    {
+        // here we enumerate all supported types.
+        case WriteEngine::WR_BYTE:
+        case WriteEngine::WR_UBYTE:
+        case WriteEngine::WR_SHORT:
+        case WriteEngine::WR_USHORT:
+        case WriteEngine::WR_INT:
+        case WriteEngine::WR_UINT:
+        case WriteEngine::WR_LONGLONG:
+        case WriteEngine::WR_ULONGLONG:
+            return currentCPInfo;
+        // all unsupported types must not be supported.
+        default:
+            return NULL; // safe choice for everything we can't do.
+    }
+}
+
+/** @brief Check if range is valid.
+ */
+static bool
+validRange(const BRM::CPInfo& cpInfo)
+{
+    return false;
+}
+
+/** @brief Let only valid ranges to be present.
+ *
+ * There can be a case that we have computed invalid range while computing updated ranges.
+ *
+ * These invalid ranges should not have CP_VALID in the status code. So, they must not
+ * come into final call to setCPInfos or something.
+ *
+ * To achieve that, we filter these invalid ranges here.
+ */
+static void
+squeezeValidCPInfos(std::vector<BRM::CPInfo>& cpInfos)
```

Comment:
I think this method is not used at all b/c of the fixed value that validRange() returns.